pkg/gds/interceptor.go

```diff
@@ -21,6 +37,7 @@ func (s *Service) serverInterceptor(ctx context.Context, in interface{}, info *g
 
 	// Call the handler to finalize the request and get the response.
 	out, err = handler(ctx, in)
+	panicked = false
```

Comment:
This is the primary panic recovery middleware. We could use the recovery middleware from grpc-ecosystem and chain things together, however we're not quite at the point where we need chained middleware and to keep our dependency footprint light, I just added the panic `recover` function. If we do move to chained middleware in the future, we should use the provided `Recovery` interceptor rather than role our own.